types/p-settle/index.d.ts

```diff
@@ -0,0 +1,28 @@
+// Type definitions for p-settle 2.0
+// Project: https://github.com/sindresorhus/p-settle#readme
+// Definitions by: Nate Silva <https://github.com/natesilva>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface SettledResult<T> {
```

Comment:
There's no way for consumers to reference this type. Can you wrap it in a namespace and have it merge with the function like so?

```ts
declare namespace pSettle {
    interface SettledResult<T> {
        // ...
    }
}

declare function pSettle<T>(input: Iterable<PromiseLike<T>>): Promise<Array<pSettle.SettledResult<T>>>;

export = pSettle;
```